bundles/org.openhab.core.config.discovery.sddp/src/main/java/org/openhab/core/config/discovery/sddp/SddpDiscoveryService.java

```diff
@@ -0,0 +1,437 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.net.CidrAddress;
+import org.openhab.core.net.NetworkAddressChangeListener;
+import org.openhab.core.net.NetworkAddressService;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a {@link DiscoveryService} implementation, which can find SDDP devices in the network.
+ * Support for bindings can be achieved by implementing and registering a {@link SddpDiscoveryParticipant}.
+ * Support for finders can be achieved by implementing and registering a {@link SddpDeviceParticipant}.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+@Component(immediate = true, service = DiscoveryService.class, configurationPid = "discovery.sddp")
+public class SddpDiscoveryService extends AbstractDiscoveryService
+        implements AutoCloseable, NetworkAddressChangeListener {
+
+    private static final int SDDP_PORT = 1902;
+    private static final String SDDP_IP_ADDRESS = "239.255.255.250";
+    private static final InetSocketAddress SDDP_GROUP = new InetSocketAddress(SDDP_IP_ADDRESS, SDDP_PORT);
+
+    private static final int READ_BUFFER_SIZE = 1024;
+    private static final Duration SOCKET_TIMOUT = Duration.ofMillis(1000);
+    private static final Duration SEARCH_LISTEN_DURATION = Duration.ofSeconds(5);
+    private static final Duration CACHE_PURGE_INTERVAL = Duration.ofSeconds(300);
+
+    private static final String SEARCH_RESPONSE_HEADER = "SDDP/1.0 200 OK";
+    private static final String ALIVE_NOTIFICATION_HEADER = "NOTIFY ALIVE SDDP/1.0";
+
+    private static final String SEARCH_REQUEST_BODY_FORMAT = """
+            SEARCH * SDDP/1.0
+            Host: "%s:%d"
+            """;
```

Comment:
I don't using Text Blocks works properly to include the necessary line breaks in the search packet. I got the active scan to work by changing this to have the \r\n line breaks:
```suggestion
    private static final String SEARCH_REQUEST_BODY_FORMAT = "SEARCH * SDDP/1.0\r\n\"Host: %s:%d\"\r\n";
            """;
```

Reply:
Aha. Yes I think Java text blocks produce \n only, whereas SDDP requires \r\n ..

Reply:
PS I can confirm that changing \n to \r\n did also fix the problem with my, (what I thought was broken), HD Powerview Hub, so it does now properly respond to the Search request.

https://github.com/openhab/openhab-addons/pull/16790#issuecomment-2133372064

Reply:
PS @mlobstein please look at the splitters in the HD Powerview Hub 'Host' header. It confirms that the last split is the Mac address, but the previous splits do not comply with your hypothesis for them.

Reply:
You can still use a Text Block, and just provide the `\r` at the end of each line. See https://www.baeldung.com/java-text-blocks section 4.2.

Reply:
I see you did, please disregard. 🙂

Reply:
I have some additional observations of SDDP packets from the DB-UB9000 that are not seen in the linked README:

When the device turns off, it sends an offline packet:

> NOTIFY OFFLINE SDDP/1.0
From: "192.168.1.2:1902"
Host: "Panasonic-dvd-80c755xxxxxx"
Type: "Panasonic:dvd"

In the network menu, there is an option to 'Send Identify', when chosen the following packet is sent:

> NOTIFY IDENTIFY SDDP/1.0
From: "192.168.1.2:1902"
Host: "Panasonic-dvd-80c755xxxxxx"
Type: "Panasonic:dvd"
Primary-Proxy: "dvd"
Proxies: "dvd"
Manufacturer: "Panasonic"
Model: "DP-UB9000"
Driver: "panasonic_dvd_dp-ub9000.c4z"